src/lib/JIT.hs

```diff
@@ -262,7 +262,12 @@ compileInstr instr = case instr of
     Heap dev -> do
       numBytes <- mul (sizeof elemTy) =<< (`asIntWidth` i64) =<< compileExpr s
       case dev of
-        CPU -> malloc     elemTy numBytes
+        CPU -> case t of
+          -- XXX: it's important to initialize pointers to zero so that we don't
+          --      try to dereference them when we serialize.
+          PtrType _ -> malloc True  elemTy numBytes
+          _         -> malloc False elemTy numBytes
```

Comment:
I find it a little worrying that this happens only in JIT, because it makes the zeroing somewhat backend specific. For example, we could implement an Imp interpreter, but it itself would have to do the zeroing too. I feel like it would be better to add a `MemZero` instruction to `Imp`.